src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc

```diff
@@ -0,0 +1,132 @@
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "can_cooperation/commands/get_interior_vehicle_data_request.h"
+#include "can_cooperation/validators/get_interior_vehicle_data_request_validator.h"
+#include "can_cooperation/validators/get_interior_vehicle_data_response_validator.h"
+#include "can_cooperation/can_module_constants.h"
+#include "can_cooperation/message_helper.h"
+#include "can_cooperation/can_module.h"
+#include "functional_module/function_ids.h"
+#include "json/json.h"
+
+namespace can_cooperation {
+
+namespace commands {
+
+using namespace json_keys;
+using namespace message_params;
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "GetInteriorVehicleDataRequest")
+
+GetInteriorVehicleDataRequest::GetInteriorVehicleDataRequest(
+    const application_manager::MessagePtr& message,
+    CANModuleInterface& can_module)
+    : BaseCommandRequest(message, can_module) {}
+
+void GetInteriorVehicleDataRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  SendRequest(functional_modules::hmi_api::get_interior_vehicle_data,
+              MessageHelper::StringToValue(message_->json_message()),
+              true);
+}
+
+void GetInteriorVehicleDataRequest::OnEvent(
+    const can_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  DCHECK_OR_RETURN_VOID(
+      (functional_modules::hmi_api::get_interior_vehicle_data == event.id()));
+  std::string info;
+
+  application_manager::Message& hmi_response = *(event.event_message());
+  bool validate_result = service()->ValidateMessageBySchema(hmi_response);
+  Json::Value value = MessageHelper::StringToValue(hmi_response.json_message());
+  std::string result_code;
+  bool success = validate_result && ParseResultCode(value, result_code, info);
+  if (success) {
+    ProccessSubscription(value);
+    response_params_[kModuleData] = value[kResult][kModuleData];
+  }
+  SendResponse(success, result_code.c_str(), info);
+}
+
+void GetInteriorVehicleDataRequest::ProccessSubscription(
+    Json::Value& hmi_response) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  Json::Value request_params;
+  Json::Reader reader;
+  reader.parse(message_->json_message(), request_params);
+
+  bool subscribe = false;
+  if (IsMember(request_params, kSubscribe)) {
+    subscribe = request_params[kSubscribe].asBool();
+  }
+  if (subscribe && !hmi_response.isMember(kIsSubscribed)) {
+    LOG4CXX_WARN(logger_,
+                 "conditional mandatory parameter "
+                     << kIsSubscribed << " missed in hmi response");
+    return;
+  }
+
+  if (!subscribe && hmi_response.isMember(kIsSubscribed)) {
+    LOG4CXX_WARN(logger_,
+                 "Parameter "
+                     << kIsSubscribed
+                     << " ignoring because it was not requested in request");
```

Comment:
@LuxoftAKutsan `ignoring because it was not requested in request` -> `is ignored due to absence in request"`